regression/pages/common/utils.py

```diff
@@ -0,0 +1,496 @@
+# -*- coding: utf-8 -*-
+
+"""
+Different utilities to be used in tests
+"""
+import re
+import time
+import functools
+import datetime
+import imaplib
+import email
+
+import requests
+from requests.auth import HTTPBasicAuth
+from edx_rest_api_client.client import EdxRestApiClient
+from opaque_keys.edx.keys import AssetKey, CourseKey
+
+from regression.pages.whitelabel.const import (
+    ACCESS_TOKEN,
+    AUTH_PASSWORD,
+    AUTH_USERNAME,
+    DEFAULT_GMAIL_USER,
+    DEFAULT_TIMEOUT,
+    ECOMMERCE_API_URL,
+    EMAIL_SENDER_ACCOUNT,
+    GMAIL_PASSWORD,
+    INITIAL_WAIT_TIME,
+    ORG,
+    TIME_OUT_LIMIT,
+    WAIT_TIME
+)
+
+
+class MailException(Exception):
+    """
+    Exceptions for Mail failures
+    """
+    pass
+
+
+class ApiException(Exception):
+    """
+    Exceptions for API failures
+    """
+    pass
+
+
+class MailClient(object):
+    """
+    Connect to Gmail clinet using imap and read mails from Inbox
+    """
+
+    def __init__(self):
+        self.mail = imaplib.IMAP4_SSL('imap.gmail.com')
```

Comment:
The client should be a variable you set somewhere, whether it's the top of the file or in the constants file. I'll re-read your arguments on why we're not using a test service.... I don't like relying on a third party for such a critical part of tests.